docs/proposals/automated-node-removal.md

```diff
@@ -0,0 +1,71 @@
+# Proposal: Automation of node removal procedure
+
+Last updated: 02.07.2021
+
+## Abstract
+
+Implementation of the option to delete CSI CRs from a node, which is going to be removed. 
+
+## Background
+
+User manual actions for node removal are described in [doc](https://github.com/dell/csi-baremetal/blob/master/docs/node-removal.md).
+The goal of this proposal is automating most of the steps of that procedure.
+
+## Proposal
+
+#### User API
+
+CSI - components on CSI side including CSI Operator
+
+Operator - application controller (or other service, which watch resources)
+
+Removal procedure:
+1. Taint a node with:
+- `node.dell.com/drain=drain:NoSchedule`
+2. Wait until Operator replaced PVs and destroyed pods
+3. Delete a node from kubernetes cluster
+4. CSI deletes all corresponding CR resources automatically (including Csibmnode)
+
+After removal the node can't be restored ad in [replacement](https://github.com/dell/csi-baremetal/blob/master/docs/proposals/node-replacement.md)! 
+
+
+## Implementation
+
+Set label:
+1. CSI Operator must watch Nodes on taint changing events
+2. Label the Csibmnode with `should-be-removed=yes` if the Node has taint `node.dell.com/drain=drain:NoSchedule`
```

Comment:
Can we use the same key value to avoid confusion: `node.dell.com/drain=drain`?

Reply:
renamed